hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/BasicMapStoreContext.java

```diff
@@ -242,7 +247,7 @@ public void run() {
     private void selectOwnedKeys(Iterable loadedKeys, MapServiceContext mapServiceContext) {
 
         initialKeys.clear();
-        int maxSizePerNode = getApproximateMaxSize(getMaxSizePerNode()) - 1;
+        int maxSizePerNode = getApproximateMaxSize(maxSizeConfig, PER_NODE) - 1;
```

Comment:
It's because eviction starts when `nodeTotalSize >= maxSize`. The -1 is not absolutely necessary but without it an entry would get evicted right away.